src/wp-includes/functions.wp-scripts.php

```diff
@@ -168,21 +168,31 @@ function wp_add_inline_script( $handle, $data, $position = 'after', $standalone
  *                                    as a query string for cache busting purposes. If version is set to false, a version
  *                                    number is automatically added equal to current installed WordPress version.
  *                                    If set to null, no version is added.
- * @param array             $args     {
+ * @param array|bool       $args     {
  *      Optional. An array of additional script loading strategies. Default empty array.
+ *      Otherwise, it may be a boolean in which case it determines whether the script is printed in the footer. Default false.
  *
  *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
+ *      @type bool      $in_footer    Optional. Whether to print the script in the footer. Default 'false'.
  * }
  * @return bool Whether the script has been registered. True on success, false on failure.
  */
 function wp_register_script( $handle, $src, $deps = array(), $ver = false, $args = array() ) {
+	if ( ! is_array( $args ) ) {
+		$args = array(
+			'in_footer' => (bool) $args,
+		);
+	}
 	_wp_scripts_maybe_doing_it_wrong( __FUNCTION__, $handle );
 
 	$wp_scripts = wp_scripts();
 
 	$registered = $wp_scripts->add( $handle, $src, $deps, $ver );
-	if ( ! empty( $args ) ) {
-		$wp_scripts->add_data( $handle, 'script_args', $args );
+	if ( ! empty( $args['in_footer'] ) ) {
+		$wp_scripts->add_data( $handle, 'group', 1 );
+	}
+	if ( ! empty( $args['strategy'] ) ) {
+		$wp_scripts->add_data( $handle, 'strategy', $args['strategy'] );
```

Comment:
I wonder if we should validate strategy values at this point, rather than in `get_intended_strategy`? If so, then we can make sure that we're only adding valid data and do less juggling down the line. Same would be applied to `wp_enqueue_script`.

Reply:
> Thanks @westonruter. I like how this simplifies things. One nitpick is that I think the unit test method, `test_get_normalized_script_args`, should likely be broken up into multiple methods or refactored to use a data provider. Either way, I think we could handle that outside the scope of this PR.

@joemcgill I too feel we can handle this outside of the scope of this PR. I agree that the test contains various _areas_ within the test case that can probably be split into smaller standalone tests (that do or do not use a data provider), but given that the changes to the test logic in this PR encapsulate and validate what @westonruter has effected in the updated logic for flattening the script_args, I feel we're good to proceed as is and open follow up test enhancements in a separate PR.

Reply:
> I wonder if we should validate strategy values at this point, rather than in `get_intended_strategy`? If so, then we can make sure that we're only adding valid data and do less juggling down the line. Same would be applied to `wp_enqueue_script`.

@joemcgill I actually agree with this. I looked into `get_intended_strategy` which is where we perform the `is_valid_strategy` check, and that does feel like the incorrect place to do so, as the offending entry point to the incorrect strategy is indeed `wp_register_script` or `wp_enqueue_script`. 

I'm happy to pick this up in this PR.

Reply:
After chatting to @westonruter and @joemcgill it was agreed that it would make sense to re-introduce the `add_data` method (removed in this PR at the time of writing this comment) and apply the `is_valid_strategy` checks therein. 

This decision is supported mainly by the following discussions/comments in the base PR:

- https://github.com/WordPress/wordpress-develop/pull/4391#discussion_r1197180090
- https://github.com/WordPress/wordpress-develop/pull/4391/files#r1179791220

Reply:
Screenshots of comments since it takes forever to load that PR now:

> [![image](https://github.com/10up/wordpress-develop/assets/134745/5e18deb0-4e6b-4869-b799-189446248ad0)](https://github.com/WordPress/wordpress-develop/pull/4391#discussion_r1197180090)

> [![image](https://github.com/10up/wordpress-develop/assets/134745/b4dce67a-e979-49c7-b424-b3a4701fc586)](https://github.com/WordPress/wordpress-develop/pull/4391/files#r1179791220)

Reply:
This is now done.

Reply:
Implemented `@dataProvider` in 7475e29